react/src/views/CreateBox/CreateBox.test.js

```diff
@@ -0,0 +1,270 @@
+import React from "react";
+import { render, waitFor, fireEvent, cleanup } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import { MockedProvider } from "@apollo/client/testing";
+import CreateBox from "./CreateBox";
+import { CREATE_BOX } from "../../utils/queries";
+import { Router } from "react-router-dom";
+import { createMemoryHistory } from "history";
+import "mutationobserver-shim";
+import { GraphQLError } from "graphql";
+
+const mocks = [
+  {
+    request: {
+      query: CREATE_BOX,
+      variables: {
+        productId: 2,
+        items: 2,
+        locationId: 2,
+        comments: "",
+        sizeId: 2,
+        qrBarcode: "387b0f0f5e62cebcafd48383035a92a",
+      },
+    },
+    result: {
+      data: {
+        createBox: {
+          id: 555,
+          box_id: 456,
+          product_id: 123,
+          items: 50,
+        },
+      },
+    },
+  },
+];
+
+describe("Renders components", () => {
+  let component = null;
+  beforeEach(() => {
+    const history = createMemoryHistory();
+    const state = { qr: "barcode=387b0f0f5e62cebcafd48383035a92a" };
+    history.push("/create-box", state);
+
+    component = render(
```

Comment:
I would strongly suggest that you create a `test-utils` file where you define this custom renderer as a function with the variables (mocks and history)

Every test will most likely need a similar renderer and you already duplicate the definition of it 5 times in this file.

Here, a few steps how to do that
https://testing-library.com/docs/react-testing-library/setup#custom-render